crates/noirc_evaluator/src/ssa.rs

```diff
@@ -87,14 +87,11 @@ pub fn create_circuit(
     enable_brillig_logging: bool,
 ) -> Result<(Circuit, DebugInfo, Abi), RuntimeError> {
     let func_sig = program.main_function_signature.clone();
+    let generated_acir = optimize_into_acir(program, enable_ssa_logging, enable_brillig_logging)?;
+    let opcodes = generated_acir.opcodes().to_vec();
```

Comment:
Why do we need to clone the opcodes here instead of destructuring them directly like we did previously?

Reply:
I see they're private now but I'd still like to avoid the clone. Maybe we can have a `take_opcodes` method which does a `std::mem::take` on them.

Reply:
I switched to a `take_opcodes` method instead of the clone